source/agora/common/BanManager.d

```diff
@@ -124,47 +96,73 @@ public class BanManager
 
     ***************************************************************************/
 
-    public this (Config config, Clock clock, string banfile_path,
-        Logger logger = Logger(ThisModule)) @safe nothrow pure
+    public this (Config config, Clock clock,
+        ManagedDatabase db, Logger logger = Logger(ThisModule)) @safe nothrow
     {
         this.config = config;
         this.log = logger;
         this.clock = clock;
-        this.banfile_path = banfile_path;
-    }
 
-    /***************************************************************************
+        this.db = db;
 
-        Load the ban data from disk
+        create_tables();
 
-    ***************************************************************************/
+        // load from cache db current state
+        this.load_whitelisted();
+        this.load_banned();
+    }
 
-    public void load ()
+    /// If the tables do not exist create them
+    private void create_tables () @trusted nothrow
     {
-        if (!this.banfile_path.exists())
-            return;  // nothing to load
+        try
+        {
+            // create banned if required
+            this.db.execute("CREATE TABLE IF NOT EXISTS banned " ~
+                "(url TEXT PRIMARY KEY, until INTEGER NOT NULL)");
```

Comment:
Just a note to future; some might encounter with this for #830 and this is OK. SQLite will store `time_t` as 64bit signed integer (32bit until 2038) and we just can't overflow this with timestamps 🙂

Reply:
So we are are ok for quite some time.